### ISSUE TITLE
Can't update over 2.27.15 using Cocoapods

### ISSUE THREAD

**Issue by dcristolovean** [aws-amplify/aws-sdk-ios#4533]
**State your question**
I cannot get the latest versions of the AWS SDK, always getting 2.27.15 with a clean pod install.
Is there a reason for that (iOS version dependency or anything else) ?  How can I update the pods to the latest version ?

**Which AWS Services are you utilizing?**
aws-amplify/aws-sdk-ios

**Provide code snippets (if applicable)**

**Environment(please complete the following information):**
 - SDK Version: 2.27,15
 - Dependency Manager: Cocoapods
 - Swift Version : whatever's in XCode 14

**Device Information (please complete the following information):**
 - Device: iPhone 11, Simulator
 - iOS Version: 16.x
 - Specific to simulators:

If you need help with understanding how to implement something in particular then we suggest that you first look into our [developer guide](https://docs.aws.amazon.com/aws-mobile/latest/developerguide/aws-mobile-android-and-iOS.html). You can also simplify your process of creating an application, as well as the associated backend setup by using the [Amplify CLI](https://aws-amplify.github.io/media/get_started).

**Comment by atierian** [comment]
Thanks for opening this issue @dcristolovean. 
You'll likely need to run `pod update` / `pod update [PODNAME]` to update to a newer version. If you already have the SDK frameworks in your `Podfile.lock`, running `pod install` will not update versions. 

See [pod install vs. pod update](https://guides.cocoapods.org/using/pod-install-vs-update.html)
> When you run pod install, it only resolves dependencies for pods that are not already listed in the Podfile.lock. 

A quick attempt at reproducing what you're experiencing confirms that `pod update` should resolve this.

```bash
>> cat Podfile
target 'tmp_sdk_ios_cocoapods' do
  use_frameworks!
  pod 'AWSMobileClient'
  pod 'AWSPinpoint'

end

>> pod install
Analyzing dependencies
Downloading dependencies
Installing AWSAuthCore (2.29.1)
Installing AWSCognitoIdentityProvider (2.29.1)
Installing AWSCognitoIdentityProviderASF (2.29.1)
Installing AWSCore (2.29.1)
Installing AWSMobileClient (2.29.1)
Installing AWSPinpoint (2.29.1)
Generating Pods project
Integrating client project

>> pod install # <-- second pod install doesn't update because versions are already pinned in Podfile.lock
Updating local specs repositories
Analyzing dependencies
Downloading dependencies
Generating Pods project
Integrating client project

>> pod update # <-- `pod update` updates to current version 
Updating local specs repositories
Analyzing dependencies
Downloading dependencies
Installing AWSAuthCore 2.30.1 (was 2.29.1)
Installing AWSCognitoIdentityProvider 2.30.1 (was 2.29.1)
Installing AWSCognitoIdentityProviderASF 2.30.1 (was 2.29.1)
Installing AWSCore 2.30.1 (was 2.29.1)
Installing AWSMobileClient 2.30.1 (was 2.29.1)
Installing AWSPinpoint 2.30.1 (was 2.29.1)
Generating Pods project
Integrating client project
```

Please let us know whether this resolves your issue. Thanks!

**Comment by dcristolovean** [comment]
I created a new empty iOS project and a new Podfile.  Here are the findings:

1.  Using only these:

  pod 'AWSMobileClient'
  pod 'AWSPinpoint'

-> Installing AWSAuthCore (2.30.1)
Installing AWSCognitoIdentityProvider (2.30.1)
Installing AWSCognitoIdentityProviderASF (2.30.1)
Installing AWSCore (2.30.1)  .....

So, with those 2 pods, I have 2.30.1

2.  Using more pods that I need:

    pod 'AWSCore'
    pod 'AWSRuntime'
    pod 'AWSAPIGateway'
    pod 'AWSPinpoint'
    pod 'AWSPluginsCore'
    pod 'AWSSNS'
    pod 'AWSS3'
    pod 'AWSCognitoIdentityProvider'
    pod 'AWSMobileClient'
    pod 'AWSAuthUI'
    pod 'AWSUserPoolsSignIn'
    pod 'AWSAppSync'

The result is completely different:

Installing AWSAPIGateway (2.26.3)
Installing AWSAppSync (3.4.2)
Installing AWSAuthCore (2.26.3)
Installing AWSAuthUI (2.26.3)
Installing AWSCognitoIdentityProvider (2.26.3)
Installing AWSCognitoIdentityProviderASF (2.26.3)
Installing AWSCore (2.26.3)
Installing AWSMobileClient (2.26.3)
Installing AWSPinpoint (2.26.3)
Installing AWSPluginsCore (1.15.6)
Installing AWSRuntime (0.0.2)
Installing AWSS3 (2.26.3)
Installing AWSSNS (2.26.3) ......

As you can see, version 2.26.3.

All are 100% clean pod installs, with the Pods folder deleted everytime, start from scratch.
Now I'm not even getting 2.27.15, but 2.26.3 in this new project.
There are clearly dependencies between those packages that prevent the latest version.  Maybe you know which one and why ?

**Comment by atierian** [comment]
We'll be cutting a new release of the AWSAppSync pod (AWS AppSync SDK for iOS) that uses the current version of the AWS SDK for iOS and follow up here once it's complete. Once that's done, you only need to do a `pod update` with your current setup.

**Comment by atierian** [comment]
AWS AppSync SDK for iOS v3.6.2 was released yesterday. Please try again and let us know if this resolved your issue, thanks!

**Comment by dcristolovean** [comment]
It seems to be fixed now, all the Pods use 2.30.1 now.

Unfortunately, I thought that would fix my other problem with S3 [(](https://github.com/aws-amplify/aws-sdk-ios/issues/4534) but it didn't, it's the same behavior

**Comment by atierian** [comment]
Thanks for letting us know. I'm closing this issue. We'll follow up regarding the S3 issue there.
